### PR TITLE
perf: configure renovate for pre-commit

### PR DESCRIPTION
**Pull request by msclock**
Anyway, unified management of dependencies is best practice.

Related #391 

Test on prs:

- https://github.com/msclock/ss-python/pull/2
- https://github.com/msclock/ss-python/pull/3

### CODE REVIEW

File: .renovaterc.json [hunk 1]
```diff
@@ -79,6 +80,16 @@
                 ]
             }
         },
+        {
+            "matchManagers": [
+                "pre-commit"
+            ],
+            "postUpgradeTasks": {
+                "commands": [
+                    "find template -type f -name '.pre-commit-config.yaml.jinja' -exec sed -i '/{{{replace \"/\" \"\\/\" depName}}}/{N;s/\\(rev: \"*{{{currentValue}}}\"*\\)/rev: {{{newValue}}}/}' {} +"
```
Comment (huxuan) [inline .renovaterc.json:89]: I would suggest have a simplier regex pattern. What do you think of '/{{{replace \"/\" \"\\/\" depName}}}/{N;s/rev: {{{currentValue}}}/rev: {{{newValue}}}/}'? Just removing the optional quote matching and unnecessary group pattern.

**Review by huxuan — COMMENTED** [review]
BTW, I may also suggest to group the update of renovate docker tag and renovate pre-commit tag. But it is OK to do it in a separate pull request since it also changes the logic in template generation.

File: .renovaterc.json (hunk 1 above)
Comment (msclock) [reply .renovaterc.json:89]: @huxuan  I will test it with `{{{replace "/" "\/" depName}}}` later. But replace `s/\\(rev: \"*{{{currentValue}}}\"*\\)` with `s/rev: {{{currentValue}}}` implies version in the field `rev: ` cannot use double quotes.

Comment (huxuan) [reply .renovaterc.json]: Actually, the changes of replace part is a typo, but I also want to confirm whether the escape is necessary.Feel free to let me know if it does not work.

**Review by huxuan — APPROVED** [review]

**Pull request merged**